File: techila/examples/python/Features/iterative_projects/mcpi_dist.py
# Copyright 2012-2013 Techila Technologies Ltd.

# This Python scrip contains the Worker Code, which will be
# distributed and sourced on the Workers. The values of the input
# parameters will be received from the parameters defined in the Local
# Control Code.

import random

def mcpi_dist(loops, jobidx, iteration):
    random.seed(jobidx * iteration)

    count = 0 # No random points generated yet, init to 0.

    i = 0
    while i < loops:
        if pow(pow(random.random(), 2) + pow(random.random(), 2), 0.5) < 1:
            count = count + 1
        i = i + 1

    return count # Return the results as a list

File: techila/examples/R/Features/streaming_callback/mcpi_dist.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This R-script contains the Worker Code, which will be distributed and sourced
# on the Workers. The values of the input parameters will be received from the
# parameters defined in the Local Control Code.
mcpi_dist <- function(loops) {

  count <- 0 # No random points generated yet, init to 0.
  for (i in 1:loops) { # Monte Carlo loop from 1 to loops
    if ((sum(((runif(1) ^ 2)  + (runif(1) ^ 2))) ^ 0.5) < 1) { # Point within the circle?
      count <- count + 1 # Increment if the point is within the circle.
    }
  }
  return(list(count=count,loops=loops)) # Return the results as a list
}
